tensorboard/webapp/widgets/linked_time_fob/linked_time_fob_controller_component.ts

```diff
@@ -194,4 +194,31 @@ export class LinkedTimeFobControllerComponent {
     // No fobs can pass the lowest step in this graph.
     return 0;
   }
+
+  stepTyped(fob: Fob, step: number) {
+    let newLinkedTime = this.linkedTime;
+    if (!newLinkedTime.end) {
+      newLinkedTime.start.step = step;
+      this.onSelectTimeChanged.emit(newLinkedTime);
+      return;
+    }
+
+    if (fob === Fob.START) {
+      newLinkedTime.start.step = step;
+      if (step > newLinkedTime.end.step) {
+        newLinkedTime.start.step = newLinkedTime.end.step;
```

Comment:
Note you could instead write 

```
      newLinkedTime = {
        start: newLinkedTime.end,
        end: newLinkedTime.start,
      };
```

And similarly on line 217.

Reply:
handled in summary comment.